HW04.Rmd

```diff
@@ -1,126 +1,186 @@
+
+# Whites
+white_intent_count <- gun_deaths %>% group_by(race, intent) %>% summarise(count=n()) %>% filter(race %in% c("White")) %>% mutate(percentage = (count / sum(count))*100) %>% arrange(desc(percentage))
+kable(white_intent_count,format="markdown")
+
+# Blacks
+black_intent_count <- gun_deaths %>% group_by(race, intent) %>% summarise(count=n()) %>% filter(race %in% c("Black")) %>% mutate(percentage = (count / sum(count))*100) %>% arrange(desc(percentage))
+kable(black_intent_count,format="markdown")
+
+# Hispanics
+hispanic_intent_count <- gun_deaths %>% group_by(race, intent) %>% summarise(count=n()) %>% filter(race %in% c("Hispanic")) %>% mutate(percentage = (count / sum(count))*100) %>% arrange(desc(percentage))
+kable(hispanic_intent_count,format="markdown")
+
+```
+
+### Are police-involved gun deaths significantly different from other gun deaths? Assess the relationship between police involvement and age, police involvement and race, and the intersection of all three variables.
+
+Think about performing stats such as a t.test to see if differences you are plotting are statistically significant
+
+(I changed the order a little here. I performed statistical test after the corresponding figure instead of doing it at the end)
+
+#### Are there differences in age depending on wether police was involved or not?
+```{r police 1}
+gun_deaths$police <- as.factor(gun_deaths$police)
+
+#Boxplots showing age in cases were police was involved and was not involved
+#Age seems to be a little bit higher for cases in which the police was not involved (0)
+ggplot(gun_deaths,mapping = aes(x = police, y= age)) +
+  geom_boxplot(,fill="gray80")+
+  theme_bw()+
+  theme(axis.title = element_text(color="black", size=12))+
+  xlab("Police involvement?")
+
+#And based on a t-test the difference appears to be statistically significant. Gun death victims are statistically older in cases in which the police was not involved.
+t.test(gun_deaths$age~gun_deaths$police)
+```
+
+#### Number of deaths per race depending on wether police was involved or not
+```{r police 2}
+#Barplots showing the number of gun deaths depending on wether police was involved or not, for each race:
+ggplot(gun_deaths, aes(x=police)) +
+    geom_bar()+
+    facet_wrap(~race,scales = "free")+
+    xlab("Police involvement?") +
+    theme_bw()
+
+#Did not really know what statistical test would be useful here!
+```
+
+#### Number of deaths per race depending on wether police was involved or not
+```{r police 3}
+#Age by race and police involment
+ggplot(gun_deaths, aes(x=police,fill=race,y=age)) +
+    geom_boxplot()+
+    xlab("Police involvement?") +
+    theme_bw()
+
+#It appears to be an effect of age and police involvement (and its interaction) on the age of gun death victims:
+aov <- aov(gun_deaths$age ~ gun_deaths$police * gun_deaths$race)
```

Comment:
Oh, ok, I just learned how to perform an ANOVA analysis, thank you! 
I did this with Mann-Whitman in a case by case basis (way slower).